homeassistant/components/advantage_air/__init__.py

```diff
@@ -50,19 +51,23 @@ async def async_get():
         update_interval=timedelta(seconds=ADVANTAGE_AIR_SYNC_INTERVAL),
     )
 
-    async def async_change(change):
-        try:
-            if await api.async_change(change):
-                await coordinator.async_refresh()
-        except ApiError as err:
-            _LOGGER.warning(err)
+    def error_handle_factory(func):
```

Comment:
Perhaps add a docstring to explain what this inline function does and why it is necessary?

Reply:
The docstring has been added. The function is a factory that returns another function wrapping the API call in a try catch.